connectors/routing/connector.go

```diff
@@ -104,23 +77,24 @@ func (rc *RoutingConnector) getConnector(scope string, namePrefix string, opName
 // if no specific scope is found,
 // RoutingConnector routes to the default scope that defined in routing config yaml file
 func (rc *RoutingConnector) _getConnector(scope, namePrefix string) (dosa.Connector, error) {
-	for _, cConfig := range rc.CConfigs {
-		if cConfig.Config.RouteTo(scope, namePrefix) {
-			return cConfig.Connector, nil
-		}
+	router, err := rc.config.FindRouter(scope, namePrefix)
+	if err != nil {
+		router, err = rc.config.FindDefaultRouter()
 	}
-	return rc._getDefaultConnector()
-}
 
-func (rc *RoutingConnector) _getDefaultConnector() (dosa.Connector, error) {
-	for _, cConfig := range rc.CConfigs {
-		if cConfig.Config.Scope == DefaultScope {
-			return cConfig.Connector, nil
-		}
+	if err != nil {
+		return nil, errors.Wrap(err, "can't find connector")
```

Comment:
This error message is not informative enough. Which connector could not be found?